src/Microsoft.Health.Fhir.Stu3.Core.UnitTests/Features/Search/Converters/NodeConverterTests/MoneyNodeToQuantitySearchValueTypeConverterTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Core.Features.Search.Converters;
+using Microsoft.Health.Fhir.Core.UnitTests.Features.Search.Converters;
+using Microsoft.Health.Fhir.ValueSets;
+using Xunit;
+using static Microsoft.Health.Fhir.Tests.Common.Search.SearchValueValidationHelper;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Stu3.Core.UnitTests.Features.Search.Converters.NodeConverterTests
+{
+    public class MoneyNodeToQuantitySearchValueTypeConverterTests : FhirNodeToSearchValueTypeConverterTests<MoneyNodeToQuantitySearchValueTypeConverter, Money>
+    {
+        [Fact]
+        public async Task GivenMoneyWithNoValue_WhenConverted_ThenNoSearchValueShouldBeCreated()
+        {
+            await Test(m => m.Value = null);
+        }
+
+        [Fact]
+        public async Task GivenMoneyWithValueCodeAndSystem_WhenConverted_ThenAQuantityValueShouldBeCreated()
+        {
+            const decimal value = 480;
+            const string code = "USD";
+
+            await Test(
+                m =>
+                {
+                    m.Value = value;
+                    m.Code = code;
+                    m.System = CurrencyValues.System;
+                },
+                ValidateQuantity,
+                new Quantity(value, code, CurrencyValues.System));
+        }
+
+        [Fact]
+        public async Task GivenMoneyWithNoCode_WhenConverted_ThenNoSearchValueShouldBeCreated()
+        {
+            const decimal value = 0.125m;
+
+            await Test(m =>
+            {
+                m.Value = value;
+                m.Code = null;
+                m.System = CurrencyValues.System;
+            });
+        }
+
+        [Fact]
+        public async Task GivenMoneyWithNoSystem_WhenConverted_ThenAQuantityValueShouldBeCreated()
+        {
+            const decimal value = 0.125m;
+            const string code = "USD";
+
+            await Test(
+                m =>
+                {
+                    m.Value = value;
+                    m.Code = code;
+                    m.System = null;
+                },
```

Comment:
`MoneyNodeToQuantitySearchValueTypeConverter` is independent of the `Money` type, but its tests are not - is this expected? Is there a way around creating a separate test file for `STU3` here?

Reply:
Something you could do is make this class abstract and leave the TTypeConverter as a generic, then have two empty classes that just specify the generic. Personally I think in a use case like that it would be ok to have all three classes in one file as two of the classes are empty.

Reply:
Awesome, thanks for the rec. Looking into this now.

Reply:
Making the change to an abstract class with a generic type doesn't end up reducing duplicate code, since `Money` has different properties in `Stu3` (all the tests would end up needing to have their version-specific implementation of the abstract class).

Reply:
Sorry, I was unclear. I meant you could combine the MoneyNode and Money test files into an abstract class and two child classes. There would still be different versions for R4, R5, and Stu3 but only one file for each instead of two.

Reply:
As discussed, we will keep the duplicate converter tests as is for now, simply because we should hopefully be removing the old converters soon